[scripts]
devserver = "python main.py --configuration-file dev-config.yml"
tests = "py.test --doctest-modules --doctest-continue-on-failure --cov-report=term-missing --cov-branch --cov=server --mysql_database=faf -o testpaths=tests -m 'not rabbitmq'"
e2e = "py.test -o testpaths=e2e_tests"
vulture = "vulture main.py server/ --sort-by-size"
doc = "pdoc3 --html --force server"

[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"

[packages]
aio_pika = "~=8.2"
aiocron = "*"
aiohttp = "*"
aiomysql = {git = "https://github.com/aio-libs/aiomysql"}
docopt = "*"
humanize = ">=2.6.0"
maxminddb = "*"
oauthlib = "*"
prometheus_client = "*"
proxy-protocol = "*"
pyjwt = {version = ">=2.4.0", extras = ["crypto"]}
pyyaml = "*"
sortedcontainers = "*"
sqlalchemy = ">=2.0.0"
trueskill = "*"
uvloop = {version = "*", markers = "sys_platform != 'win32'"}

[dev-packages]
hypothesis = "*"  # Versions between 6.47.1 and 6.56.4 added a prerelease dependency. See https://github.com/pypa/pipenv/issues/1760
pdoc3 = "*"
pytest = "*"
pytest-asyncio = "*"
pytest-cov = "*"
pytest-mock = "*"
vulture = "*"

[requires]
python_version = "3.10"
